### DISCUSSION TITLE
Gabor Patch Plugin

### DISCUSSION THREAD

**Discussion by robfuscator** [jspsych/jsPsych#2665] [Show and tell]
Dear community

I'm working on a plugin to generate gabor patches and use them in a jsPsych experiment. The stimuli can be customized and even embedded in dynamic noise. You can find it here:

https://github.com/kogpsy/jspsych-gabor-stimulus-plugin

I'm posting this here to (1) maybe receive some feedback, and to (2) ask whether this could be included as a core plugin? As of now, it seems to be working pretty good in my use cases, it's extensively documented but not at all covered with tests.

Any opinions?

Best regards,
Robin

**Comment by jodeleeuw** [comment]
This looks great! 🤩 

I love the inclusion of `generateNoise` and `generateNoiseFrames`. Have you thought about exposing `generateGaborPatch` so that folks could use the stimulus in plugins that display arbitrary HTML content?

There are some stylistic differences with the plugins in the core library. For example, the `choices` array is nested under the `config` object, and the parameters use camel case. I think it would be helpful to bring the style into alignment with the rest of the library if we include it in the main repo.

**Reply by robfuscator** [reply]
I'm glad you like it! :slightly_smiling_face: 

No, I haven't thought about exposing `generateGaborPatch`, but I think it's a great idea!

Regarding the stylistic differences: it would surely make sense to align the plugin with your defaults. I chose to use a `config` object to enable consumers to use TypeScript to get autocomplete and some syntax checking mechanism, but I could just extract the `choices` array and use _snake_case_ for the parameter names. What do you think about that?

Also, how would I go about including it in the main repo, once the above issues are resolved, and some tests are implemented? Could I just create a pull request with my repo as a new `package`?
